src/blobs.rs

```diff
@@ -15,6 +15,11 @@ pub struct Collection {
 
 impl Collection {
     pub(crate) fn new(blobs: Vec<Blob>, total_blobs_size: u64) -> Self {
+        let mut blobs = blobs;
+        let n = blobs.len();
+        blobs.sort_by(|a, b| a.name.cmp(&b.name));
+        blobs.dedup_by(|a, b| a.name == b.name);
+        debug_assert_eq!(n, blobs.len());
```

Comment:
what if I have two blobs of the same name?

Reply:
I don't want to support that. Do you?

Reply:
OK, I guess it would not too bad. It would make this basically a `BTreeMap<String, BTreeSet<Hash>>`. But still, do you have a specific use case in mind?

Reply:
no, I just don’t want inconsistent panics 😅 if we don’t want duplicates I would expect this to error with a matching error message and not panic

Reply:
Ah, ok. That is reasonable. I think it is ensured upward of this code that names are distinct, hence the assert! and not returning anyhow::Result.

Reply:
Alright, implemented this as a runtime check in the deep dirs branch...